core/src/apps/cardano/sign_tx.py

```diff
@@ -747,15 +857,36 @@ def _validate_output(
 
     if address_parameters := output.address_parameters:
         if signing_mode != CardanoTxSigningMode.ORDINARY_TRANSACTION:
+            # change outputs are allowed only in ORDINARY_TRANSACTION
             raise INVALID_OUTPUT
 
         validate_output_address_parameters(address_parameters)
         _fail_if_strict_and_unusual(address_parameters)
+        address_type = address_parameters.address_type
     elif output.address is not None:
         validate_output_address(output.address, protocol_magic, network_id)
+        address_type = get_address_type(get_address_bytes_unsafe(output.address))
     else:
         raise INVALID_OUTPUT
 
+    if output.datum_hash is not None:
+        if signing_mode not in (
```

Comment:
Well, we thought this [here](https://github.com/trezor/trezor-firmware/pull/1851/files#diff-7c83c159b0b0767810cbafb1fe4883e35aeb1922ef67dbcfc49a47c9dd0482db) as well, and then an unexpected signing mode came and we forgot to change the condition.

I think it's safest to always "whitelist" signing modes, ie. if there is a new one, we have to manually allow it everywhere instead of assuming that an option will be allowed in all future signing modes.

> If there ever is a new signing mode, chances are datum_hash will be allowed for that too so this list would grow.

I prefer this over accidentaly allowing `datum_hash` on a new signing mode where it shouldn't be allowed.

Reply:
Good point 👍  Let's keep this.